### DISCUSSION TITLE
Receive DMX data

### DISCUSSION THREAD

**Discussion by hrueger** [node-dmx/dmx#116] [Ideas]
Hi,
are there any plans to add the functionality to receive DMX data?
I'd create a PR if you like the idea.

Would be interesting to know what drivers could support this.

**Comment by wiedi** [comment]
At least the Enttec USB DMX PRO devices could support this. I don't think any of my fixtures support being queried though, that's why I never added this.

I'm not sure about how the interface for this should look like, so I'd be interested in your PR proposal.

**Reply by hrueger** [reply]
Cool. I'll think about it and make a PR.
Artnet and sACN should support it, too.

**Reply by hrueger** [reply]
>I don't think any of my fixtures support being queried though, that's why I never added this.

I just read that sentence 😅 Must have missed it last time.

I didn't think about querying fixtures, but rather reading an incoming universe, for example from a connected DMX controller. The Enttec USB DMX Pro has one Universe in and two out and Artnet / sACN should work, too. The `sACN` lib in the `sACN` driver was even built primarily for receiving and not sending.

**Reply by wiedi** [reply]
Ah, at least conceptually that should be a lot easier.

Would be good to have an example API consumer of such an interface to validate the design.
